nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/energy/AngleTreeGenerator.java

```diff
@@ -426,56 +437,51 @@ public void measureAtomTree(ITree itree, List<List<Atom>> atomTree) {
+                for (Bond bond : bondMap.get(atom)) {
+                    atom.addBond(bond);
+                }
+            }
+
+            ringClosures = new HashMap<>();
+            for (Bond bond : closureBonds) {
+                bond.begin.addBond(bond);
+                bond.end.addBond(bond);
+                addRingClosureSet(ringClosures, bond.begin, bond.end);
+            }
+            if (itree instanceof Molecule) {
+                mol = (Molecule) itree;
+                mol.updateAtomArray();
+                mol.resetGenCoords();
+                mol.setupRotGroups();
+                mol.genCoords();
             }
-        }
-        ringClosures = new HashMap<>();
-        for (Bond bond : closureBonds) {
-            bond.begin.addBond(bond);
-            bond.end.addBond(bond);
-            addRingClosureSet(ringClosures, bond.begin, bond.end);
-        }
-        if (itree instanceof Molecule) {
-            mol = (Molecule) itree;
-            mol.updateAtomArray();
-            mol.resetGenCoords();
-            mol.setupRotGroups();
-            mol.genCoords();
         }
     }
-//   List<Atom> atomPathList = new ArrayList<>();
-//        // Adds all the ring closures for bonds broken in rings
-//        ringClosures = new HashMap<>();
-//        for (MNode mNode : pathNodes) {
-//            if (!mNode.isRingClosure()) {
-//                atomPathList.add(mNode.getAtom());
-//            } else {
-//                Atom atom1 = mNode.getAtom();
-//                Atom atom2 = mNode.getParent().getAtom();
-//                if ((ringClosures.containsKey(atom1) && ringClosures.get(atom1).containsKey(atom2)) || (ringClosures.containsKey(atom2) && ringClosures.get(atom2).containsKey(atom1))) {
-//                } else {
-//                    addRingClosure(atom1, atom2);
-//                    addRingClosurePairs(atom1, atom2);
-//                    addRingClosurePairs(atom2, atom1);
-//                }
-//            }
-//        }
 
     public static void dumpAtomTree(List<List<Atom>> atomTree) {
         for (List<Atom> branch : atomTree) {
 
             for (Atom atom : branch) {
                 System.out.printf("%8s", atom == null ? "____" : atom.getShortName());
             }
-            System.out.println(" " + branch.get(2).rotUnit + " " + branch.get(2).rotActive);
+            for (int i=branch.size();i<8;i++) {
+                System.out.printf("%8s", "");
+            }
+            Atom branchAtom = branch.get(2);
+            System.out.printf(" %3d %6b %3d ", branchAtom.rotUnit, branchAtom.rotActive, branchAtom.irpIndex);
+            for (int i = 3;i<branch.size();i++) {
+                branchAtom = branch.get(i);
+                System.out.printf("%6s %8.2f %8.2f %8.2f",branchAtom.getName(), Math.toDegrees(branchAtom.valanceAngle), branchAtom.bondLength, Math.toDegrees(branchAtom.dihedralAngle));
+            }
+            System.out.println();
```

Comment:
we should change these print statements to logs since we are trying to remove the print statements

Reply:
Switched to logs (though still a method which uses StringBuilder and can return string, I call this from Python while working with new structures)